src/__test__/test_production_4.py

```diff
@@ -0,0 +1,162 @@
+import unittest
+import itertools as it
+
+from graph import Graph
+from model import Node, NodeAttrs, EdgeAttrs, Edge
+
+from production import P4
+
+class TestProduction2(unittest.TestCase):
```

Comment:
```suggestion
class TestProduction4(unittest.TestCase):
```

Reply:
I added a commit fixing test class names and changing assertions to what we used during class